atc/compression/compression.go

```diff
@@ -0,0 +1,19 @@
+package compression
+
+import (
+	"io"
+
+	"github.com/concourse/baggageclaim"
+)
+
+//go:generate counterfeiter . Compression
+
+type Compression interface {
+	NewReader(io.ReadCloser) (Reader, error)
+	Encoding() baggageclaim.Encoding
+}
+
+type Reader interface {
```

Comment:
This is literally [`io.ReadCloser`](https://golang.org/pkg/io/#ReadCloser)

```
type ReadCloser interface {
    Reader
    Closer
}

type Reader interface {
    Read(p []byte) (n int, err error)
}

type Closer interface {
    Close() error
}
```

which you used as the argument for `NewReader` :thinking:  was there a specific reason for having a separate iface?

Reply:
Because the zstd Decoder has a different Close() method who doesn't return error

https://github.com/klauspost/compress/blob/4a2d40e4b07e5b3d333bc0569facd0f2dbf4ef39/zstd/decoder.go#L372

Reply:
Good point I will update.